pgxn/neon/communicator.c

```diff
@@ -576,7 +646,7 @@ readahead_buffer_resize(int newsize, void *extra)
 	 * iteration on the dataset, and trivial compaction.
 	 */
 	for (end = MyPState->ring_unused - 1;
-		 end >= MyPState->ring_last && end != UINT64_MAX && nfree != 0;
+		 end >= MyPState->ring_last && end != InfiniteXLogRecPtr && nfree != 0;
```

Comment:
Incorrect change: This is a slot index we're checking (against overflow), not an LSN. Same with a few other such cases.

Reply:
Undone